packages/account-sdk/src/sign/base-account/EphemeralSigner.ts

```diff
@@ -0,0 +1,251 @@
+import { Communicator } from ':core/communicator/Communicator.js';
+import { standardErrors } from ':core/error/errors.js';
+import { RPCRequestMessage, RPCResponseMessage } from ':core/message/RPCMessage.js';
+import { RPCResponse } from ':core/message/RPCResponse.js';
+import { AppMetadata, ProviderEventCallback, RequestArguments } from ':core/provider/interface.js';
+import {
+  logHandshakeCompleted,
+  logHandshakeError,
+  logHandshakeStarted,
+  logRequestCompleted,
+  logRequestError,
+  logRequestStarted,
+} from ':core/telemetry/events/scw-signer.js';
+import { parseErrorMessageFromAny } from ':core/telemetry/utils.js';
+import { SDKChain, createClients } from ':store/chain-clients/utils.js';
+import { correlationIds } from ':store/correlation-ids/store.js';
+import { store } from ':store/store.js';
+import {
+  decryptContent,
+  encryptContent,
+  exportKeyToHexString,
+  importKeyFromHexString,
+} from ':util/cipher.js';
+import { SCWKeyManager } from './SCWKeyManager.js';
+
+type ConstructorOptions = {
+  metadata: AppMetadata;
+  communicator: Communicator;
+  callback: ProviderEventCallback | null;
+};
+
+/**
+ * EphemeralSigner is designed for single-use payment flows.
+ *
+ * Key differences from Signer:
+ * 1. Maintains isolated instance state instead of relying on global store
+ * 2. Cleanup only clears instance-specific state (key manager)
+ * 3. Does NOT clear global store state (chains, accounts, etc.)
+ * 4. Optimized for one-shot operations like wallet_sendCalls and wallet_sign
+ *
+ * This prevents:
+ * - Race conditions when multiple ephemeral flows run concurrently
+ * - Pollution of global store state by ephemeral operations
+ * - Accidental clearing of shared chain client configurations
+ */
+export class EphemeralSigner {
+  private readonly communicator: Communicator;
+  private readonly keyManager: SCWKeyManager;
```

Comment:
hmm i think keyManager could still be an issue
* keyManager is still reading and writing to global state in localStorage
* keyManager.clear() rotate the global key pair, which may intervene with concurrent operation

Reply:
Good call. Added createStoreInstance() that can create either persistent (localStorage) or ephemeral (in-memory) store instances.

Now Signer/EphemeralSigner: Both accept a storeInstance parameter and pass it down to KeyManager